src/client/index.ts

```diff
@@ -41,7 +41,7 @@ export class ShardedCounter<ShardsKey extends string> {
    * @param name The key to update the counter for.
    * @param count The amount to increment the counter by. Defaults to 1.
    */
-  async add<Name extends string = ShardsKey>(
+  async add<Name extends ShardsKey>(
```

Comment:
this means (I believe) that you can no longer pass in a custom string if you've defined shards in your config

Reply:
correct, but i think that's actually good.

if you explicitly give the generic, you get the same behavior as before

```ts
const counter = new ShardedCounter<string>(components.shardedCounter, {
  shards: { users: 10 },
});
```

and if you have a fixed set of keys, it protects against typos

```ts
const counter = new ShardedCounter(components.shardedCounter, {
  shards: { users: 10, friends: 20 },
});

await counter.inc(ctx, "frends");
```

Reply:
yeah that's a good call. In fact what we probably want is for shards to either be keyed or fixed, so `{ shards: 10 }` or `{ shards: { users: 10 } }` vs. also having `defaultShards`